File: cluster/hello.py
#!/usr/bin/env python3
"""hello.py - Task 1: kiem tra mpi4py chay tren 1 may.
Chay: mpirun -np 4 python3 hello.py
"""
from mpi4py import MPI

comm = MPI.COMM_WORLD
rank = comm.Get_rank()
size = comm.Get_size()
node = MPI.Get_processor_name()

print(f"Hello tu process {rank} / {size} tren may {node}")

File: cluster/hello.c
/* hello.c - Task 1: kiem tra OpenMPI chay tren 1 may.
 * Bien dich:  mpicc hello.c -o hello
 * Chay:       mpirun -np 4 ./hello
 */
#include <mpi.h>
#include <stdio.h>

int main(int argc, char **argv) {
    MPI_Init(&argc, &argv);

    int rank, size, len;
    char node[MPI_MAX_PROCESSOR_NAME];
    MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    MPI_Comm_size(MPI_COMM_WORLD, &size);
    MPI_Get_processor_name(node, &len);

    printf("Hello tu process %d / %d tren may %s\n", rank, size, node);

    MPI_Finalize();
    return 0;
}
